compiler/rustc_typeck/src/check/fn_ctxt.rs

```diff
@@ -453,6 +453,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     ]) => {
                         // A reborrow has no effect before a dereference.
                     }
+                    (&[Adjustment { kind: Adjust::Pointer(PointerCast::ReifyFnPointer), .. }],
+                     &[Adjustment { kind: Adjust::Pointer(PointerCast::NotConstFnPointer), .. }])  => { }
```

Comment:
This arm should extend `*entry.get_mut()` with adj instead of letting it be overriden below.